files/common/etc/systemd/system/systemd-networkd.service.d/override.conf

```diff
@@ -20,3 +20,12 @@
 #
 [Service]
 Environment=SYSTEMD_LOG_LEVEL=debug
+
+#
+# We use "netplan apply" to make network changes which causes
+# a restart of the systemd-networkd service. Since we can
+# have any number of network configuration changes, we may exceed
+# the service's default restart limit causing the service to
+# fail. To avoid this we disable the restart limit here.
+#
+StartLimitBurst=0
```

Comment:
Should we put some reasonable limit in here to avoid having the service restart endlessly in case of failure?

Reply:
I'm open to a number but I'm not sure what that should be. Our unit tests generate tons of configuration changes and will be the first to hit the limit. I'm also not sure if it's better to have the service restart continuously or end up in a failed state. Either way, I'm happy to put a number here. Any suggestions?

Reply:
Anything that is above our unit test limit would do. Do you think something like 20 would be sufficient? Even 100 would be better than unlimited. The reason I don't like unlimited restart is that it fills up the logs fairly quickly and makes it harder to diagnose the original failure.

Reply:
I ran some tests and using a value of 30 was enough to allow the network unit tests to pass. I will use that value.